src/cmd/api/main.go

```diff
@@ -87,6 +87,13 @@ func main() {
 	// Register service on consul (only for local env)
 	registerService(cfg, discoveryClient)
 
+	// Run watcher on services, it updates the transformation service cache if a new service
+	// is register/deregister on consul
+	go func() {
+		err := discoveryClient.Watch()
+		log.Fatal("Discovery client watcher crash : ", err)
```

Comment:
Nope ! It can also gracefully terminate, this log is not good (you can simply remove it or replace by a INFO level log saying that the watcher stoped.